cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/OperationsDashboardHttpHandlerTest.java

```diff
@@ -222,46 +228,85 @@ public void testDashboardWithNamespaceFiltering() throws Exception {
-    // calculate the expected number of scheduled runs according to the query duration and the schedules
-    long expectedScheduledProgram1 = durationSecs / TimeUnit.MINUTES.toSeconds(sched1Mins);
-    long expectedScheduledProgram2 = durationSecs / TimeUnit.MINUTES.toSeconds(sched2Mins);
+    List<Long> runTimesSchedule1 =
+      getExpectedRuntimes(TimeUnit.SECONDS.toMinutes(currentTime),
+                          TimeUnit.SECONDS.toMinutes(endTime), sched1Mins, currentTime);
+
+    Set<DashboardProgramRunRecord> expectedRunRecords = new HashSet<>();
+    String userId = impersonator.getUGI(sched1.getProgramId()).getUserName();
+    for (long runTime : runTimesSchedule1) {
+      expectedRunRecords.add(
+        new DashboardProgramRunRecord(sched1.getProgramId().getNamespace(),
+                                      ArtifactSummary.from(ARTIFACT_ID1.toApiArtifactId()),
+                                      new DashboardProgramRunRecord.ApplicationNameVersion(
+                                        sched1.getProgramId().getApplication(), sched1.getProgramId().getVersion()),
+                                      sched1.getProgramId().getType().name(), sched1.getProgramId().getProgram(),
+                                      null, userId, "SCHEDULED", runTime, null, null, null, null, null));
+    }
     // assert the number of scheduled runs are expected for both programs
-    Assert.assertEquals(expectedScheduledProgram1,
-                        dashboardRecords.stream()
-                          .filter(record -> SCHEDULED_PROG1_ID.getProgram().equals(record.getProgram())).count());
-    Assert.assertEquals(expectedScheduledProgram2,
-                        dashboardRecords.stream()
-                          .filter(record -> SCHEDULED_PROG2_ID.getProgram().equals(record.getProgram())).count());
-    // TODO: compare the actual DashboardProgramRunRecord objects after
-    // https://github.com/caskdata/cdap/pull/10254 is merged
-
-    // assert the artifact id is correct
-    Assert.assertTrue(dashboardRecords.stream()
-                        .filter(record -> SCHEDULED_PROG1_ID.getProgram().equals(record.getProgram()))
-                        .allMatch(schedule -> ArtifactSummary.from(ARTIFACT_ID1.toApiArtifactId()).equals(
-                          schedule.getArtifact())));
-    Assert.assertTrue(dashboardRecords.stream()
-                        .filter(record -> SCHEDULED_PROG2_ID.getProgram().equals(record.getProgram()))
-                        .allMatch(schedule -> ArtifactSummary.from(ARTIFACT_ID2.toApiArtifactId()).equals(
-                          schedule.getArtifact())));
+    List<DashboardProgramRunRecord> schedule1Runs = dashboardRecords.stream()
+      .filter(record -> SCHEDULED_PROG1_ID.getProgram().equals(record.getProgram())).collect(Collectors.toList());
+    Assert.assertEquals(expectedRunRecords.size(), schedule1Runs.size());
+    Assert.assertTrue(schedule1Runs.containsAll(expectedRunRecords));
```

Comment:
Since we check the size equality first and then do a containsAll, can't both the checks be replaced by an equals check? Is it because of ordering?

Reply:
updated to use list, since we know schedules will be ordered by time.